src/uController.SourceGenerator/MinimalCodeGenerator.cs

```diff
@@ -570,7 +673,7 @@ private void GenerateTryParse(MethodInfo tryParseMethod, string sourceName, stri
                 }
                 else
                 {
-                    WriteLine($"if ({sourceName} == null)");
+                    WriteLine($"if (string.IsNullOrEmpty({sourceName}))");
```

Comment:
Is this right?

Reply:
I believe so. Reason for this is the discrepancy between what you get out of `ResolveByQueryOrRoute` when no parameter is provided. For the query scenario, you'll get an empty `StringValue`. For the route scenario, you'll get `null` so the previous scenario was only handling the route scenario successfully.